From b46144a54a14e00b024bd323fc0f0101151e1277 Mon Sep 17 00:00:00 2001
From: Basanagouda Koppad <basanagoudax.n.koppad@intel.com>
Date: Wed, 6 Jul 2022 00:21:25 +0530
Subject: [PATCH 1/6] Fixed fd double close in mesa3d library.

fixed double close issue in iris_screen_destroy() function.
Tracked-On: OAM-102839
Signed-of-by: Shekhar Chauhan <shekhar.chauhan@intel.com>

---
 src/gallium/drivers/iris/iris_screen.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/gallium/drivers/iris/iris_screen.c b/src/gallium/drivers/iris/iris_screen.c
index fc9b732708c..f1a98bb309e 100644
--- a/src/gallium/drivers/iris/iris_screen.c
+++ b/src/gallium/drivers/iris/iris_screen.c
@@ -625,7 +625,8 @@ iris_screen_destroy(struct iris_screen *screen)
    u_transfer_helper_destroy(screen->base.transfer_helper);
    iris_bufmgr_unref(screen->bufmgr);
    disk_cache_destroy(screen->disk_cache);
-   close(screen->winsys_fd);
+   // fd close is handled by owner module
+   //close(screen->winsys_fd);
    ralloc_free(screen);
 }
 
-- 
2.17.1

